d3a_interface/utils.py

```diff
@@ -454,3 +456,60 @@ def calculate_home_area_stats_from_repr_dict(cls, representation):
             if cls.is_home_area(area)
         ]
         return len(devices_per_home), mean(devices_per_home) if devices_per_home else None
+
+
+def sort_list_of_dicts_by_attribute(input_list: List[Dict],
+                                    attribute: str,
+                                    reverse_order=False):
+    """Sorts a list of dicts by a given attribute.
+
+    Args:
+        input_list: List[Dict]
+        attribute: attribute to sort against
+        reverse_order: if True, the returned list will be sorted in descending order
+
+    Returns: List[Dict]
+
+    """
+
+    return sorted(input_list,
+                  key=lambda obj: obj.get(attribute),
+                  reverse=reverse_order)
+
+
+def perform_pay_as_bid_match(market_offers_bids_list_mapping):
```

Comment:
I can't find usages of this method outside of the `myco-api-client` repository. If that's the case, I would think that this function might belong there, rather than in `d3a-interface`. What do you think? Are there other usages I couldn't find?

Reply:
The d3a has a perform_pay_as_bid function but with a slightly different interface, this is why I ported this here
In the context of my next task I may create an `Adapter` in the d3a and remove the function from there